internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -342,165 +342,198 @@ func (a *Adapter) EnsureIntegrationPipelineRunsExist() (controller.OperationResu
 // EnsureGlobalCandidateImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the Snapshot is created
 func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResult, error) {
+	if !a.shouldUpdateGlobalCandidateList() {
+		return controller.ContinueProcessing()
+	}
+
+	var err error
+	if gitops.IsComponentSnapshot(a.snapshot) {
+		err = a.updateComponentSnapshot()
+	} else if gitops.IsOverrideSnapshot(a.snapshot) {
+		err = a.updateOverrideSnapshot()
+	}
+
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	// Mark the Snapshot as already added to global candidate list to prevent it from getting added again when the Snapshot
+	// gets reconciled at a later time
+	err = gitops.MarkSnapshotAsAddedToGlobalCandidateList(a.context, a.client, a.snapshot, "The Snapshot's component(s) was/were added to the global candidate list")
+	if err != nil {
+		a.logger.Error(err, "Failed to update the Snapshot's status to AddedToGlobalCandidateList")
+		return controller.RequeueWithError(err)
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// shouldUpdateGlobalCandidateList checks if the snapshot should update the global candidate list
+func (a *Adapter) shouldUpdateGlobalCandidateList() bool {
 	if !gitops.IsComponentSnapshotCreatedByPACPushEvent(a.snapshot) && !gitops.IsOverrideSnapshot(a.snapshot) {
 		a.logger.Info("The Snapshot was neither created for a single component push event nor override type, not updating the global candidate list.")
-		return controller.ContinueProcessing()
+		return false
 	}
 
 	if gitops.IsSnapshotMarkedAsAddedToGlobalCandidateList(a.snapshot) {
 		a.logger.Info("The Snapshot's component was previously added to the global candidate list, skipping adding it.")
-		return controller.ContinueProcessing()
+		return false
 	}
 
+	return true
+}
+
+// updateComponentSnapshot updates global candidate list for component snapshots
+func (a *Adapter) updateComponentSnapshot() error {
```

Comment:
nit: maybe a more descriptive name would help when following the code, something like `updateGCLForComponentSnapshot` or similar.
That's if I'm reading the code right.